file/file_prefetch_buffer.cc

```diff
@@ -771,6 +771,12 @@ bool FilePrefetchBuffer::TryReadFromCache(const IOOptions& opts,
 bool FilePrefetchBuffer::TryReadFromCacheUntracked(
     const IOOptions& opts, RandomAccessFileReader* reader, uint64_t offset,
     size_t n, Slice* result, Status* status, bool for_compaction) {
+  // We disallow async IO for compaction reads since they are performed in
+  // the background anyways and are less latency sensitive compareed to
+  // user-initiated reads
+  (void)for_compaction;
```

Comment:
With the change to unify the logic, `for_compaction` is not used outside of this `assert` so we need this `(void) for_compaction` to prevent the compiler from complaining